java/flight/flight-integration-tests/src/main/java/org/apache/arrow/flight/integration/tests/FlightSqlScenario.java

```diff
@@ -0,0 +1,136 @@
+ * and that the Arrow schemas are returned as expected.
+ */
+public class FlightSqlScenario implements Scenario {
+
+  @Override
+  public FlightProducer producer(BufferAllocator allocator, Location location) throws Exception {
+    return new FlightSqlScenarioProducer(allocator);
+  }
+
+  @Override
+  public void buildServer(FlightServer.Builder builder) throws Exception {
+
+  }
+
+  @Override
+  public void client(BufferAllocator allocator, Location location, FlightClient client)
+      throws Exception {
+    final FlightSqlClient sqlClient = new FlightSqlClient(client);
+
+    validateMetadataRetrieval(sqlClient);
+
+    validateStatementExecution(sqlClient);
+
+    validatePreparedStatementExecution(sqlClient, allocator);
+  }
+
+  private void validateMetadataRetrieval(FlightSqlClient sqlClient) throws Exception {
+    final CallOption[] options = new CallOption[0];
+
+    validate(FlightSqlProducer.Schemas.GET_CATALOGS_SCHEMA, sqlClient.getCatalogs(options),
+        sqlClient);
+    validate(FlightSqlProducer.Schemas.GET_SCHEMAS_SCHEMA,
+        sqlClient.getSchemas("catalog", "db_schema_filter_pattern", options),
+        sqlClient);
+    validate(FlightSqlProducer.Schemas.GET_TABLES_SCHEMA,
+        sqlClient.getTables("catalog", "db_schema_filter_pattern", "table_filter_pattern",
+            Arrays.asList("table", "view"), true, options), sqlClient);
+    validate(FlightSqlProducer.Schemas.GET_TABLE_TYPES_SCHEMA, sqlClient.getTableTypes(options),
+        sqlClient);
+    validate(FlightSqlProducer.Schemas.GET_PRIMARY_KEYS_SCHEMA,
+        sqlClient.getPrimaryKeys(TableRef.of("catalog", "db_schema", "table"), options),
+        sqlClient);
+    validate(FlightSqlProducer.Schemas.GET_EXPORTED_KEYS_SCHEMA,
+        sqlClient.getExportedKeys(TableRef.of("catalog", "db_schema", "table"), options),
+        sqlClient);
+    validate(FlightSqlProducer.Schemas.GET_IMPORTED_KEYS_SCHEMA,
+        sqlClient.getImportedKeys(TableRef.of("catalog", "db_schema", "table"), options),
+        sqlClient);
+    validate(FlightSqlProducer.Schemas.GET_CROSS_REFERENCE_SCHEMA,
+        sqlClient.getCrossReference(TableRef.of("pk_catalog", "pk_db_schema", "pk_table"),
+            TableRef.of("fk_catalog", "fk_db_schema", "fk_table"), options),
+        sqlClient);
+    validate(FlightSqlProducer.Schemas.GET_SQL_INFO_SCHEMA,
+        sqlClient.getSqlInfo(new FlightSql.SqlInfo[] {FlightSql.SqlInfo.FLIGHT_SQL_SERVER_NAME,
+            FlightSql.SqlInfo.FLIGHT_SQL_SERVER_READ_ONLY}, options), sqlClient);
+  }
+
+  private void validateStatementExecution(FlightSqlClient sqlClient) throws Exception {
+    final CallOption[] options = new CallOption[0];
+
+    validate(FlightSqlScenarioProducer.getQuerySchema(),
+        sqlClient.execute("SELECT STATEMENT", options), sqlClient);
+
+    IntegrationAssertions.assertEquals(sqlClient.executeUpdate("UPDATE STATEMENT", options),
+        10000L);
```

Comment:
Just a nit, but can we move these numbers to constants so it's clear what they represent? (`UPDATE_STATEMENT_EXPECTED_ROWS` or something)

Reply:
(The same goes for the producer, C++, etc.)

Reply:
Thanks! Done